src/gui/guiFormSpecMenu.cpp

```diff
@@ -2487,9 +2513,51 @@ bool GUIFormSpecMenu::parseStyle(parserData *data, const std::string &element, b
 	}
 
 	if (style_type) {
-		theme_by_type[selector] |= spec;
+		theme_by_type[selector].push_back(spec);
 	} else {
-		theme_by_name[selector] |= spec;
+		theme_by_name[selector].push_back(spec);
+	}
+
+	// Backwards-compatibility for existing _hovered/_pressed properties
+	if(spec.hasProperty(StyleSpec::BGCOLOR_HOVERED) || spec.hasProperty(StyleSpec::BGIMG_HOVERED) || spec.hasProperty(StyleSpec::FGIMG_HOVERED)) {
```

Comment:
Please mark those enum entries as deprecated in `StyleSpec.h` so that it's clear they're only there for compatibility purposes.

Also: Missing space after `if`. Lines too long. Please split after `||` for a nice looking list.